Sources/Plasma/CoreLib/hsStream.cpp

```diff
@@ -230,7 +230,7 @@ bool hsStream::AtEnd()
 
 bool hsStream::IsTokenSeparator(char c)
 {
-    return (isspace(c) || c==',' || c=='=');
+    return (isspace(static_cast<unsigned char>(c)) || c==',' || c=='=');
```

Comment:
??  ctype functions (`isspace`, `isalnum`, etc) take an `int`, which both a signed and unsigned implementation of `char` should automatically and safely promote to...  Why all the casts?

Reply:
That's the problem. `char` is signed and gets sign-extended to `int`, but the ctype functions are [only defined for `unsigned char` values and `EOF`](https://en.cppreference.com/w/cpp/string/byte/isspace). See also e. g. https://nullprogram.com/blog/2023/02/11/#character-classification-and-mapping

MSVCRT has a debug assertion ensuring that you don't pass any other negative values. That started failing once I typed non-ASCII characters into the console, so I went ahead and fixed the issue for all ctype calls (there weren't all that many).

Reply:
Ah, that make sense...  Although perhaps it makes more sense to test for that specific case, since characters outside the 0-127 range will always be UTF-8 multibyte sequence bytes when coming from ST::strings.

Reply:
Not sure what you mean exactly... something like `c < 0x80 && isspace(c)`? I don't think that's necessary if `c` is already `char`-sized. We don't use locales, so all the `is` functions will return `false` for all bytes outside the ASCII range.

(I *did* use `c < 0x80` checks where `c` is a larger type like `wchar_t` or `uint32_t`, because just a cast isn't sufficient there.)